java.lang.ClassCircularityError: jdk/internal/misc/VirtualThreads thrown by grpc-virtual-executor

### <!-- Component -->

<!-- Zeebe- -->

### <!-- Affected version -->

<!-- -8.6 -->

### <!-- Severity -->

<!-- High- -->

### <!-- Likelihood -->

<!-- Unknown_ -->

### Description

The following error is thrown by the grpc-virtual-executor thread: 
```
io.grpc.internal.SerializingExecutor - Exception while executing runnable io.grpc.internal.ServerImpl$ServerTransportListenerImpl$1Ha
ndleServerCall@71e0114c
java.lang.IllegalStateException: java.lang.ClassCircularityError: jdk/internal/misc/VirtualThreads
```

When this occurs, the gateway stops responding to gRPC requests, but the `/actuator/health/liveness` shows that service is up and `zbctl status` shows a healthy gateway.

Full error stacktrace:

```
2025-05-27 07:13:49.592 [] [grpc-virtual-executor] [] ERROR
      io.grpc.internal.SerializingExecutor - Exception while executing runnable io.grpc.internal.ServerImpl$ServerTransportListenerImpl$1Ha
ndleServerCall@71e0114c
java.lang.IllegalStateException: java.lang.ClassCircularityError: jdk/internal/misc/VirtualThreads
        at io.grpc.internal.ServerImpl$ServerTransportListenerImpl$1HandleServerCall.runInternal(ServerImpl.java:617) ~[grpc-core-1.68.1.ja
r:1.68.1]
        at io.grpc.internal.ServerImpl$ServerTransportListenerImpl$1HandleServerCall.runInContext(ServerImpl.java:603) ~[grpc-core-1.68.1.j
ar:1.68.1]
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.68.1.jar:1.68.1]
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.68.1.jar:1.68.1]
        at java.base/java.util.concurrent.ThreadPerTaskExecutor$TaskRunner.run(Unknown Source) ~[?:?]
        at java.base/java.lang.VirtualThread.run(Unknown Source) ~[?:?]
Caused by: java.lang.ClassCircularityError: jdk/internal/misc/VirtualThreads
```

### Steps to reproduce

Not yet known.

### Current behavior

When this occurs, the gateway stops responding to gRPC requests, but the `/actuator/health/liveness` shows that service is up and `zbctl status` shows a healthy gateway.

### Expected behavior

Possible solution ideas:
- the error doesn't occur at all
- or the error leads to an unhealthy gateway
- or the error causes the gateway to shut down

### Environment

SM

### Version

- Component Version: 8.6.5


### Rootcause

_No response_

### Solution ideas

Possible solution ideas:
- the error doesn't occur at all
- or the error leads to an unhealthy gateway
- or the error causes the gateway to shut down

### Dev -> QA handover

- Resources:
- Versions to validate:
- Release version (in which version this feature will be released):


### Links

- https://jira.camunda.com/browse/SUPPORT-27299

--- a/zeebe/util/src/main/java/io/camunda/zeebe/util/error/VirtualMachineErrorHandler.java
+++ b/zeebe/util/src/main/java/io/camunda/zeebe/util/error/VirtualMachineErrorHandler.java
@@ -11,9 +11,10 @@ import java.lang.Thread.UncaughtExceptionHandler;
 import org.slf4j.Logger;
 
 /**
- * Handles all Throwables and exits for {@link VirtualMachineError}. It can also be used as a {@link
- * UncaughtExceptionHandler uncaught exception handler}, for example as the {@link
- * Thread#setDefaultUncaughtExceptionHandler default uncaught exception handler}
+ * Handles all Throwables and exits for {@link VirtualMachineError} and other virtual machine errors
+ * like {@link ClassCircularityError}. It can also be used as a {@link UncaughtExceptionHandler
+ * uncaught exception handler}, for example as the {@link Thread#setDefaultUncaughtExceptionHandler
+ * default uncaught exception handler}
  */
 public final class VirtualMachineErrorHandler
     implements FatalErrorHandler, UncaughtExceptionHandler {
@@ -38,7 +39,7 @@ public final class VirtualMachineErrorHandler
    */
   @Override
   public void handleError(final Throwable e) {
-    if (e instanceof VirtualMachineError) {
+    if (e instanceof VirtualMachineError || e instanceof ClassCircularityError) {
       tryLogging(e);
       System.exit(EXIT_CODE);
     }

